### DISCUSSION TITLE
Getting "Permission denied" error while installing ruby-2.5.8 on M1 Mac

### DISCUSSION THREAD

**Discussion by rnhurt** [rbenv/ruby-build#1903] [Build failures] [answered]
I have a brand new M1 MacBook Pro that I'm trying to install Ruby 2.5.8 on.  I used to use rvm but thought I would give rbenv a try on this new machine.  So, I installed rbenv with Homebrew (`brew install rbenv`) , but the `rbenv init` command in my `.zshrc` file, and opened a new shell.  There were no errors and everything looked great; until I tried to install Ruby 2.5.8.

- macOS Monterey v12.1
- `ruby -v`  "ruby 2.6.8p205 (2021-07-07 revision 67951) [universal.arm64e-darwin21]"
- `rbenv -v` "rbenv 1.2.0"
- `ruby-build --version` "ruby-build 20211227"

```
➜  rbenv install 2.5.8
Downloading openssl-1.1.1l.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/0b7a3e5e59c34827fe0c3a74b7ec8baef302b98fa80088d7f9153aa16fa76bd1
Installing openssl-1.1.1l...
Installed openssl-1.1.1l to /Users/richard.hurt/.rbenv/versions/2.5.8

Downloading ruby-2.5.8.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.5/ruby-2.5.8.tar.bz2
Installing ruby-2.5.8...

WARNING: ruby-2.5.8 is past its end of life and is now unsupported.
It no longer receives bug fixes or critical security updates.

mkdir: /usr/local/opt/mysql@5.7/lib: Permission denied

BUILD FAILED (macOS 12.1 using ruby-build 20211227)

Inspect or clean up the working tree at /var/folders/qv/q3_1vc7n4pgczm8s_9_b40v00000gn/T/ruby-build.20220111035615.84408.L3YpNj
Results logged to /var/folders/qv/q3_1vc7n4pgczm8s_9_b40v00000gn/T/ruby-build.20220111035615.84408.log

...
```

It looks like it can't create a new directory in `/usr/local`, and I think the latest version of macOS restricts activity in /usr/local (for some reason).
```
➜   ll /usr/local
total 0
drwxrwxr-x  8 root  wheel   256B Jan 10 23:28 bin
drwxrwxr-x  3 root  wheel    96B Nov  4 16:40 lib
drwxrwxr-x  6 root  wheel   192B Jan  5 09:39 sessionmanagerplugin
drwxrwxr-x  3 root  wheel    96B Nov 14 05:14 share
```

I was going to try to force the issue by creating `/usr/local/opt` manually, but I didn't figure that was the correct solution and macOS gave me some pushback on that solution.  So, what is the proper solution to this issue?  Do I manually create the `/usr/local/opt` directory?  If so, with what permissions?  Homebrew seemed to solve this issue by using `/opt/homebrew` for everything.  Is that something "ruby-build" should do?

**Comment by mislav** [comment, marked as answer]
Hi, installing Ruby with rbenv and ruby-build absolutely does not need write permissions to `/usr/local`. Also, compiling Ruby does not link it to mysql automatically. Something is wrong here.

Can you check your environment variables for the mention of "mysql"? `env | grep mysql`

Can you share with us the contents of the full log file? You can paste it as a Gist and link it here.

**Reply by rnhurt** [reply]
I think you figured it out!  I had copied my `.zshrc` file from my old computer and it had some flags set for compiling MySQL.  Once I removed those flags the error went away.

Thanx!

**Reply by siegy22** [reply]
FYI had the same weird thing going on but had to remove the `~/.viminfo` file on Kali Linux. The CPP and LD_FLAGS were in there somehow? :thinking:
